hieroplatform/src/main/java/org/hiero/sketches/HLogLog.java

```diff
@@ -0,0 +1,111 @@
+package org.hiero.sketches;
+
+import org.hiero.table.api.IColumn;
+import org.hiero.table.api.IMembershipSet;
+import org.hiero.table.api.IRowIterator;
+import org.hiero.utils.HashUtil;
+
+/**
+ * A class that computes an approximation of the number of distinct elements in a column. Elements
+ * are identified via their hashcode. The class uses the HyperLogLog algorithm for large estimates
+ * and LinearCounting algorithm for small estimates.
+ */
+public class HLogLog {
+    private final int regNum; //number of registers
+    private final int logregNum;
+    private final byte[] registers;
+    private final int seed;
+
+    /**
+     * @param logRegNum the logarithm of the number of registers. should be in 4...16
+     *        Going beyond 16 would make the data structure big for no good reason.
+     *        Setting logRegNum in at 10-12 should give roughly 2% accuracy most of the time
+     */
+    public HLogLog(int logRegNum, int seed) {
+        if (!spaceValid(logRegNum))
+            throw new IllegalArgumentException("HLogLog initialized with number out of range");
+        this.regNum = 1 << logRegNum;
+        this.registers = new byte[this.regNum];
+        this.logregNum = logRegNum;
+        this.seed = seed;
+    }
+
+    /**
+     * adds the int 'value' to the data structure.
+     * Uses the first bits to identify the register and then counts trailing zeros
+     * @param value already assumed to be a random hash of the item
+     */
+    private void add(int value) {
```

Comment:
It will lose accuracy at around million. I've thought of some ways to (maybe) provide accurate estimates all the way up to a few million, but even if it works it will not go beyond that.